tests/ha/priority_fencing_delay.pm

```diff
@@ -0,0 +1,72 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test the priority fencing delay feature
+# The node with the master resource must always win the fencing match
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use hacluster;
+
+sub stonith_iptables {
+    my ($self, $count, $cluster) = @_;
+    my $partner_ip = is_node(1) ? get_ip(choose_node(2)) : get_ip(choose_node(1));
+
+    while ($count ne 0) {
+        script_run "iptables -A INPUT -s $partner_ip -j DROP; iptables -A OUTPUT -d $partner_ip -j DROP";
+        if (is_node(1)) {
+            # Wait for the stonith match, then flush the rules for the next test
+            sleep 20;
```

Comment:
I'm guessing this `sleep` is here to avoid flushing the iptables rules before the other node is fenced, so I am wondering if this can be changed for something that checks the status of the other node or the state of the cluster before continuing. Probably something like `script_run "until $crm_mon_cmd | grep -qi offline ; do sleep 1; done", 60`. Only worry would be that the `offline` status of the other node it's so fast that `crm_mon` in this node fails to see it.

Reply:
Exactly, same remark as Loic, I'm on it!